extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/generate/StockMethodsAdder.java

```diff
@@ -187,6 +199,24 @@ private void generateSave(ClassCreator classCreator, FieldDescriptor entityClass
         }
     }
 
+    private boolean isPersistable(DotName entityDotName) {
+        ClassInfo classInfo = index.getClassByName(entityDotName);
+        if (classInfo == null) {
+            throw new IllegalStateException("Entity " + entityDotName + " was not part of the Quarkus index");
+        }
+
+        if (classInfo.interfaceNames().contains(DotNames.SPRING_DATA_PERSISTABLE)) {
+            return true;
+        }
+
+        DotName superDotName = classInfo.superName();
+        if (superDotName.equals(DotNames.OBJECT)) {
+            return false;
+        }
+
+        return isPersistable(superDotName);
```

Comment:
Iteration is more efficient than recursion :wink:

Reply:
No doubt 👍